lib/router/src/router.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+
+import { Link, Location, navigate, LocationProvider } from '@reach/router';
+import { ToggleVisibility } from './visibility';
+import { queryFromString, storyDataFromString } from './utils';
+
+interface RenderData {
+  path: string;
+  location: object;
+  navigate: (to: string) => void;
+  viewMode?: string;
+  storyId?: string;
+}
+interface MatchingData {
+  match: null | { path: string };
+}
+
+interface QueryLocationType {
+  children: (a: RenderData) => React.ReactNode;
+}
+interface QueryMatchType {
+  path: string;
+  startsWith: boolean;
+  children: (a: MatchingData) => React.ReactNode;
+}
+interface RouteType {
+  path: string;
+  startsWith: boolean;
+  hideOnly: boolean;
+  children: (a: RenderData) => React.ReactNode;
+}
+
+const queryNavigate = (to: string) => {
+  navigate(`?path=${to}`);
+};
+
+const QueryLink = ({ to, children, ...rest }: { to: string; children: React.ReactNode }) => (
```

Comment:
Use type for `{ to: string; children: React.ReactNode }`

Reply:
✅